src/main/scala/org/bitcoins/core/protocol/script/ScriptPubKey.scala

```diff
@@ -26,14 +29,21 @@ sealed trait ScriptPubKey extends NetworkElement with BitcoinSLogger {
    */
   lazy val asm : Seq[ScriptToken] = ScriptParser.fromBytes(bytes)
 
+  def p2shAddress(network : NetworkParameters) : P2SHAddress = {
```

Comment:
Don't think this is the right place for this, it should be inside of `P2SHScriptPubKey`. Perhaps we could add a `address` function here, and then implement that function inside of all of the sub classes, however we would then have to figure out a way to generate an address for a `NonStandardScriptPubKey` -- i don't think this behavior is defined in Bitcoin Core.